### PR TITLE
Exercise 7 temp

### PR DESCRIPTION
**Pull request by vaibhav1sh**
Pull request for Module 7 exercise.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+script:
+- echo "hello world"
```
Comment (JackMeadDev) [inline .travis.yml]: We can remove this line now 😄

File: todo_app/trello_api_calls.py [hunk 2]
```diff
@@ -76,6 +76,8 @@ def create_trello_board():
         'name': os.environ.get('TRELLO_BOARD_NAME_CREATE')
     }
     url = api_prefix + '1/boards/'
+    # TESTING
+    print('TRELLO BOARD NAME',os.environ.get('TRELLO_BOARD_NAME_CREATE'))
```
Comment (JackMeadDev) [inline todo_app/trello_api_calls.py]: We can delete these now

File: README.md [hunk 3]
```diff
@@ -110,11 +110,37 @@ Production Image: (application is copied to image as well, hence no need to use
...
 
 The .drawio files can be edited through online tool available at https://app.diagrams.net/
+
+## **Module-7**
+
+### Pre-requisites 
+Use following commands to create test image.
+
+```bash
+$ docker run todo-app:test tests/test_viewmodel.py
```
Comment (JackMeadDev) [inline README.md]: I think you've copied the wrong command in here; this is one of your run commands (not caught below) but I think you meant to have a build command here

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+- docker build --target development --tag todo-app:dev .
+- docker build --target production --tag todo-app:prod .
```
Comment (JackMeadDev) [inline .travis.yml]: Worth noting that there's no major gain from building these images here; we'd usually stick to building when we're actually going to use the image e.g. when we're ready to push the image to a repository & deploy it to a testing environment which we'll add in as part of module 8

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- docker build --target test --tag todo-app:test .
+- docker run todo-app:test tests/test_viewmodel.py
+- docker run --env-file .env.test todo-app:test tests/test_app.py
+- docker run --env TRELLO_BOARD_NAME_CREATE=E2E_test_board --env TRELLO_KEY=${TRELLO_KEY}
```
Comment (JackMeadDev) [inline .travis.yml]: Docker has a short syntax when you're passing environment variables that have the same name inside & outside the container;
`docker run TRELLO_BOARD_NAME_CREATE=E2E_test_board --env TRELLO_KEY --env TRELLO_TOKEN ...`

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+script:
+- echo "hello world"
+- docker build --target development --tag todo-app:dev .
+- docker build --target production --tag todo-app:prod .
+- docker build --target test --tag todo-app:test .
+- docker run todo-app:test tests/test_viewmodel.py
+- docker run --env-file .env.test todo-app:test tests/test_app.py
```
Comment (JackMeadDev) [inline .travis.yml:4]: This setup is OK for now, but will become more painful as you add more testing files as you'll need to add a new command for each. What we might do is instead group any that require fake credentials, and run those with `docker run --env-file .env.test todo-app:test tests` and then group the ones with live credentials `docker run --env-file .env todo-app:test live_tests`  (or some better name!)

File: Dockerfile [hunk 7]
```diff
@@ -15,12 +15,33 @@ RUN poetry config virtualenvs.create false --local
...
+RUN LATEST=`curl -sSL \
+https://chromedriver.storage.googleapis.com/LATEST_RELEASE` && \
+echo "Installing chromium webdriver version ${LATEST}" && curl -sSL \
+https://chromedriver.storage.googleapis.com/${LATEST}/chromedriver_linux64.zip \
+-o chromedriver_linux64.zip &&  apt-get install unzip -y &&\
+unzip ./chromedriver_linux64.zip
+COPY todo_app todo_app
+COPY tests tests
+COPY tests_e2e tests_e2e
+ENV PYTHONPATH "${PYTHONPATH}:/app"
```
Comment (JackMeadDev) [inline Dockerfile:46]: Worth noting that you can skip this line if you add an empty `__init__.py` file to your tests/tests_e2e folders which is good practice to help Python determine modules

**Review by JackMeadDev — APPROVED** [review]
Nice work, I can see your Travis pipeline is passing and was able to easily run your tests inside docker locally. I've left a few thoughts below, and in particular your Readme has a mistake that we should clear up, but happy for you to merge this when you're ready

File: .travis.yml (hunk 6 above)
Comment (vaibhav1sh) [reply .travis.yml:4]: I agree - never thought of that! Will add to my list - to be addressed as a stretch goal later in the program.

File: Dockerfile (hunk 7 above)
Comment (vaibhav1sh) [reply Dockerfile:46]: Thanks - I will add this to my to-do list, will address as a stretch goal later in the program.

File: .travis.yml (hunk 1 above)
Comment (vaibhav1sh) [reply .travis.yml]: Done.

File: todo_app/trello_api_calls.py (hunk 2 above)
Comment (vaibhav1sh) [reply todo_app/trello_api_calls.py]: Thanks - removed these as well.

File: README.md (hunk 3 above)
Comment (vaibhav1sh) [reply README.md]: I did ! Thank you for pointing that out - corrected.

File: .travis.yml (hunk 4 above)
Comment (vaibhav1sh) [reply .travis.yml]: Removed the un-necessary builds

File: .travis.yml (hunk 5 above)
Comment (vaibhav1sh) [reply .travis.yml]: Thanks - I have added these changes.

**Comment by vaibhav1sh** [comment]
Merging after review.

**Pull request merged**